### PR TITLE
Optimizations & Improvements

### PR DESCRIPTION
**Pull request by Moire9**
this nut

### CODE REVIEW

File: src/main/java/com/proudmuslim/andromeda/mixin/PlayerEntityMixin.java [hunk 1]
```diff
@@ -13,18 +10,17 @@
 import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
-// Mixins HAVE to be written in java due to constraints in the mixin system.
+/** Mixins HAVE to be written in java due to constraints in the Mixin system. */
 @Mixin(PlayerEntity.class)
 public abstract class PlayerEntityMixin extends LivingEntity {
 
-    protected PlayerEntityMixin(EntityType<? extends LivingEntity> type, World world) {
+    protected PlayerEntityMixin(final EntityType<? extends LivingEntity> type, final World world) {
```
Comment (ChachyDev) [inline src/main/java/com/proudmuslim/andromeda/mixin/PlayerEntityMixin.java:17]: que?

File: src/main/kotlin/com/proudmuslim/andromeda/Andromeda.kt [hunk 2]
```diff
@@ -13,25 +14,34 @@ import net.minecraft.item.Item
...
+
+    /**
+     * Register a block as both a block and an item.
+     *
+     * @param[block] the block
+     * @param[_namespace] the namespace under which you want to register the block (i.e. mod name)
+     * @param[_path] item name or something I dunno IntelliJ's not letting me see what [Identifier.path] is used in
+     * @param[inventoryGroup] group to be sorted under in the creative inventory
+     */
+    private fun dualRegister(block: Block /* Insert the name of the class you used to set up the block there */, _namespace: String, _path: String, inventoryGroup: ItemGroup ) {
```
Comment (ChachyDev) [inline src/main/kotlin/com/proudmuslim/andromeda/Andromeda.kt]: can't you just name that register

File: src/main/kotlin/com/proudmuslim/andromeda/CreeperHeartItem.kt [hunk 3]
```diff
@@ -8,12 +8,18 @@ import net.minecraft.item.Item
 import net.minecraft.util.Hand
 
 class CreeperHeartItem(settings: Settings): Item(settings) {
+
+    private var using: Boolean = true
```
Comment (ChachyDev) [inline src/main/kotlin/com/proudmuslim/andromeda/CreeperHeartItem.kt:12]: que?

File: src/main/java/com/proudmuslim/andromeda/mixin/PlayerEntityMixin.java (hunk 1 above)
Comment (Moire9) [reply src/main/java/com/proudmuslim/andromeda/mixin/PlayerEntityMixin.java:17]: final = speed

Comment (Moire9) [reply src/main/java/com/proudmuslim/andromeda/mixin/PlayerEntityMixin.java:17]: also i think it's in my style settings to do that and i ran an automatic thing

File: src/main/kotlin/com/proudmuslim/andromeda/CreeperHeartItem.kt (hunk 3 above)
Comment (Moire9) [reply src/main/kotlin/com/proudmuslim/andromeda/CreeperHeartItem.kt:12]: fires twice so we only make it fire every other time it's called

File: src/main/kotlin/com/proudmuslim/andromeda/Andromeda.kt (hunk 2 above)
Comment (Moire9) [reply src/main/kotlin/com/proudmuslim/andromeda/Andromeda.kt]: shhhhhhh

File: src/main/kotlin/com/proudmuslim/andromeda/CreeperHeartItem.kt (hunk 3 above)
Comment (ChachyDev) [reply src/main/kotlin/com/proudmuslim/andromeda/CreeperHeartItem.kt:12]: ah

**Comment by proudmuslim-dev** [comment]
Pog

**Comment by proudmuslim-dev** [comment]
Did not close on accident shh

**Pull request merged**